### PR TITLE
Add type definitions for netlify-identity-widget

### PR DESCRIPTION
**Pull request by naveenkumarsangi**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by nkprince007** [comment]
As suggested by the package maintainer (@sw-yx) of netlify-identity-widget, the type definitions are expected to be maintained here.

File: types/netlify-identity-widget/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Type definitions for netlify-identity-widget 1.4
+// Project: https://github.com/netlify/netlify-identity-widget
+// Definitions by: Naveen Kumar Sangi <https://github.com/nkprince007>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export { };
```
Comment (uniqueiniquity) [inline types/netlify-identity-widget/index.d.ts]: This line is unnecessary.
In order to have an ES6 style, callable import statement when consuming these type definitions, the consumer must enable `--esModuleInterop` (or more specifically, `--allowSyntheticDefaultImports`), thereby allowing them to use `import NetlifyIdentityWidget from "netlify-identity-widget";` in the same way that one would use `import NetlifyIdentityWidget = require("netlify-identity-widget");`

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by nkprince007** [comment]
Thanks for the early review @uniqueiniquity. The response has been extremely timely. 👍

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**